src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -169,7 +169,7 @@ protected NodeContext GetNode(string msbuildLocation, string commandLineArgs, in
             {
                 var candidateProcesses = GetPossibleRunningNodes(msbuildLocation);
 
-                CommunicationsUtilities.Trace("Attempting to connect to each existing msbuild.exe process in turn to establish node {0}...", nodeId);
+                CommunicationsUtilities.Trace("Attempting to connect to each existing {1} process in turn to establish node {0}...", nodeId, Path.GetFileName(msbuildLocation));
```

Comment:
I don't think msbuildLocation is guaranteed to have a value at this point, so you might want to add default behavior.

Reply:
Maybe change `GetPossibleRunningNodes` to return `(expectedProcessName, candidateProcesses)` and log that? That'll be helpful for the .NET Core host case too.

Reply:
Fixed!

Reply:
Whoops, didn't see the most recent comment.  I'll get on that.

Reply:
Using the name gathered from `GetPossibleRunningNodes` worked out better in this case.